src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+    private final long defaultDelay;
+    /** The ticker used to schedule this job. */
+    private final Ticker ticker;
+    /** The executor of this background job. */
+    private final Executor executor;
+
+    /** Tiny job to run on the ticker, invoking the execution of the {@link #realJob} on the
+     * executor (only set in {@code WAITING} state). */
+    private Runnable waitingTickerJob = null;
+    /** Running state of this job. */
+    private JobState state = JobState.IDLE;
+    /** Job execution thread, only if the job is running (only set in {@code RUNNING} state). */
+    private Thread thread = null;
+    /** Next absolute time we have a job execution scheduled, or {@link #NO_EXECUTION} if none. */
+    private long nextExecutionTime = NO_EXECUTION;
+
+    /**
+     * Constructs a delayed background job with the given default delay. Negative delays
+     * are treated as zero delay.
+     * The {@link Ticker} given and its {@link Executor} <b>must</b> have an asynchronous
+     * implementation of respectively
+     * {@link Ticker#queueTimedJob(Runnable, String, long, boolean, boolean) queueTimedJob} and
+     * {@link Executor#execute(Runnable, String) execute}. When this background job is
+     * {@link BackgroundJob#terminate() terminated}, the running job will be notified by means of
+     * interruption of its thread. Hence, the job implementer must take care not to swallow
+     * {@link InterruptedException}.
+     * @param job the job to run in the background
+     * @param name a human-readable name for the job
+     * @param delay the default background job aggregation delay in milliseconds
+     * @param ticker an asynchronous ticker with asynchronous executor
+     *
+     * @see TickerDelayedBackgroundJobFactory
+     */
+    public TickerDelayedBackgroundJob(Runnable job, String name, long delay, Ticker ticker) {
+        if (job == null || name == null || ticker == null || ticker.getExecutor() == null) {
+            throw new NullPointerException();
+        }
+        if (delay < 0) {
+            delay = 0;
+        }
+        this.realJob = new DelayedBackgroundRunnable(job);
+        this.name = name;
+        this.defaultDelay = delay;
+        this.ticker = ticker;
+        this.executor = ticker.getExecutor();
+    }
+
+    /**
+     * Triggers scheduling of the job with the default delay if no job is scheduled. If a job
+     * is already scheduled later than the default delay, it is rescheduled at the default delay.
+     *
+     * The first trigger received after the start of the last job execution leads to scheduling of
+     * another execution of the job, either after the default delay or when the currently executing
+     * job is finished, whichever comes last. A newly constructed delayed background job can be
+     * assumed to have started its last job infinitely in the past.
+     * @see #triggerExecution(long)
+     */
+    @Override
+    public synchronized void triggerExecution() {
```

Comment:
FYI: You can use inheritDoc AND still put other specific JavaDoc before/after it. It will concatenate both.

About the other stuff: Can confirm that both have been fixed recently.

-> Done.